examples/plot_score_alg.py

```diff
@@ -0,0 +1,189 @@
+"""
+.. _ex-score-algorithm:
+
+==============================================================================================
+Order based algorithms for causal discovery from observational data without latent confounders
+==============================================================================================
+
+We will simulate some observational data from a Structural Causal Model (SCM) and
+demonstrate how we will use the SCORE algorithm. The example can be easily adapted
+to CAM, DAS and  NoGAM methods.
+
+CAM :footcite:`Buhlmann2013`, SCORE :footcite:`rolland2022`, DAS :footcite:`Montagna2023a`,
+NoGAM :footcite:`Montagna2023b` algorithms perform causal discovery in a two steps procedure.
+Given observational i.i.d. data from an Additive Noise Model without
+latent confounders, first the method estimates a topological ordering of the causal variables.
+This partial oredring can be represented as a fully connected graph, where every node has an
+incoming edge from all its predecessors in the ordering. Second, the resulting fully connected
+DAG is pruned by some variable selection procedure.
+Note that knowing the topological ordering is already sufficient for estimating causal effects.
+Nevertheless, the pruning step is justified by the fact that operating with a sparser graph is
+statistically more efficient.
+
```

Comment:
It would be helpful to have some brief description as well here about how one estimates the topological order. Then have a brief description on generally what pruning involves. The previous paragraphs are useful because it is now clear to me that what I am observing are a combination of methods that differ in one of the two step procedures.

Reply:
Yeah that would work!